### ISSUE TITLE
Linux arm64 ilasm not encoding value type constraints as expected

### ISSUE THREAD

**Issue by jaredpar** [dotnet/runtime#44440]
As a part of a customer request I was updating dotnet/roslyn to build and test on Linux ARM64. This mostly comes down to us consuming the Linux ARM64 version of `ilasm` as the rest of our repository is tooling architecture neutral. 

Once I make that change though I found that a number of our unit test, about a dozen, are failing and they all have roughly the same pattern: 

```il
.class public V3<valuetype ([mscorlib]System.ValueType) T> { }
```

The type parameter encoded here does not round trip as having the value type constraint as recognized by the compiler. Other examples are `B1.M2` below. 

```il
.class public B1 extends class A<object>
{
  .method public virtual instance void M2<valuetype (object)U>() { ret }
}
```

The compiler examples are more complicated but are best illustrated by the following two tests:

- `ObjectAndValueTypeMetadataConstraints`: https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Test/Symbol/Symbols/GenericConstraintTests.cs#L5210
- `ObjectAndValueTypeMethodMetadataConstraints`: https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Test/Symbol/Symbols/GenericConstraintTests.cs#L5239

Have not dug further to see what changed about the IL (hard to dump the IL in the remote session I'm in). 

Seems odd that this would be an `ilasm` change between architectures but the only real change here on our side is running on ARM64. The rest of our tests pass so it's not a fundamental issue we have on ARM64 loading metadata. Seems specific to this pattern type.

**Comment by briansull** [comment]
@jaredpar 
Is there a full repro case, ready to present to IlAsm/IlDasm for this?

**Comment by briansull** [comment]
Does IlDasm step add any ..param constraint directives?

**Comment by briansull** [comment]
This is fixed with #44850
